drivers/interrupt_controller/intc_loapic.c

```diff
@@ -200,7 +200,7 @@ uint32_t z_loapic_irq_base(void)
  *
  * This associates an IRQ with the desired vector in the IDT.
  */
-__boot_func
+__pinned_func
```

Comment:
This function is also called from `z_irq_controller_irq_config` , which is marked as `__boot_func`. Wouldn't this patch cause problems for it? Maybe that one could also be moved outside boot section, but I guess you'd have to re-check all its callers and so on.

Reply:
no problem. boot function can call pinnned function.
Initially both boot and pinned section are loaded into SRAM and their corresponding pages are marked as present.